java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/jobexecution/CompactSortedFiles.java

```diff
@@ -223,6 +232,24 @@ private RecordsProcessed compactSplittingByCopy() throws IOException, StateStore
         return new RecordsProcessed(recordsProcessed, recordsProcessed);
     }
 
+    private RecordsProcessed compactSplittingByReference() throws StateStoreException {
+        Map<String, FileInfo> activeFilesByPartitionIdAndFilename = stateStore.getActiveFiles().stream()
+                .collect(Collectors.toMap(file -> file.getPartitionId() + "|" + file.getFilename(), Function.identity()));
+        List<FileInfo> inputFileInfos = compactionJob.getInputFiles().stream()
+                .map(filename -> compactionJob.getPartitionId() + "|" + filename)
+                .map(activeFilesByPartitionIdAndFilename::get)
+                .collect(Collectors.toUnmodifiableList());
+        List<FileInfo> outputFileInfos = new ArrayList<>();
+        for (FileInfo inputFileInfo : inputFileInfos) {
+            for (String childPartitionId : compactionJob.getChildPartitions()) {
+                outputFileInfos.add(SplitFileInfo.referenceForChildPartition(inputFileInfo, childPartitionId));
```

Comment:
Can we add a log line here as so that we can see what's going on? In the logs for a splitting compaction job you see the compaction job itself, and then a bunch of stuff from the state store but nothing about the compaction job itself.

Reply:
Done, thanks.